src/transformers/models/llama/modeling_llama.py

```diff
@@ -865,6 +838,9 @@ def _init_weights(self, module):
             if module.padding_idx is not None:
                 module.weight.data[module.padding_idx].zero_()
 
+    def _setup_cache(self, cache_cls, max_batch_size):
+        for layer in self.model.layers:
+            layer.self_attn.past_key_value = cache_cls(self.config, max_batch_size, device=layer.self_attn.o_proj.weight.device)
```

Comment:
well, if we ever will have other new cache classes in the future, it's fine here and also having `ALL_CACHE_CLASSES` in `utils.py`.